common/src/main/java/com/commercetools/sunrise/payment/domain/PaymentServiceProvider.java

```diff
@@ -0,0 +1,48 @@
+package com.commercetools.sunrise.payment.domain;
+
+import io.sphere.sdk.payments.Payment;
+import io.sphere.sdk.payments.PaymentMethodInfo;
+import io.sphere.sdk.payments.PaymentStatus;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+/**
+ * A payment service provider provides functions that can be used to handle payment related tasks in
+ * a PSP specific way.
+ * @author mgatz
+ */
+public interface PaymentServiceProvider {
+
+    /**
+     * @return a unique identifier of the payment service provider
+     */
+    String getId();
+
+    /**
+     * @return a list of payment methods this PSP supports
+     */
+    List<PaymentMethodInfo> getAvailablePaymentMethods(); // TODO decide parameters
```

Comment:
I am not sure which value such function has if i have to provide methods and filter (yes its a part of the shop logic) them also. I think it should at least provide a list of available payment methods. In the future there will be endpoint for that but for now i would put it as custom object with container ```payment``` and key ```methods``` and value as JSON list.